src/WinRT.Runtime/Projections/SystemTypes.cs

```diff
@@ -55,21 +55,26 @@ public struct DateTimeOffset
         // 'System.DateTimeOffset.Ticks' is relative to 01/01/0001
         public long UniversalTime;
 
+        // Numer of ticks counted between 0001-01-01, 00:00:00 and 1601-01-01, 00:00:00.
+        // You can get this through:  (new DateTimeOffset(1601, 1, 1, 0, 0, 1, TimeSpan.Zero)).Ticks;
+        private const long ManagedUtcTicksAtNativeZero = 504911232000000000;
+        // DO NOT use ToFileTime/FromFileTime, which don't support negative UniversalTime.
+
         public struct Marshaler
         {
             public DateTimeOffset __abi;
         }
 
         public static Marshaler CreateMarshaler(global::System.DateTimeOffset value)
         {
-            return new Marshaler { __abi = new DateTimeOffset { UniversalTime = value.ToFileTime() } };
+            return new Marshaler { __abi = new DateTimeOffset { UniversalTime = value.Ticks - ManagedUtcTicksAtNativeZero } };
         }
 
         public static DateTimeOffset GetAbi(Marshaler m) => m.__abi;
 
         public static global::System.DateTimeOffset FromAbi(DateTimeOffset value)
         {
-            return global::System.DateTimeOffset.FromFileTime(value.UniversalTime);
+            return new global::System.DateTimeOffset(value.UniversalTime + ManagedUtcTicksAtNativeZero, global:: System.TimeSpan.Zero);
         }
 
         public static unsafe void CopyAbi(Marshaler arg, IntPtr dest) =>
```

Comment:
Probably want to do something similar below in FromManaged as it uses ToFileTime.

Reply:
Thanks @huoyaoyuan for addressing the other comments.  I am not sure whether you missed this one.  Right below this function in `FromManaged`, we still use `ToFileTime` as seen below.  That probably should be changed to reflect the same changes you did in `CreateMarshaler`.  Once that is done, I can merge this PR.

```
public static DateTimeOffset FromManaged(global::System.DateTimeOffset value)
{
    return new DateTimeOffset { UniversalTime = value.ToFileTime() };
}
```